include/shmem_pack_helper.cuh

```diff
@@ -0,0 +1,228 @@
+#pragma once
+#ifdef NVSHMEM_COMMS
+#include <cooperative_groups.h>
+namespace cg = cooperative_groups;
+
+template <int dest, typename Arg> __device__ inline void *getShmemBuffer(int shmemindex, const Arg &arg)
```

Comment:
Can you add doxygen for the functions in this file?

Reply:
sure